src/Core/Client.vala

```diff
@@ -794,3 +800,33 @@ public class AppCenterCore.Client : Object {
         return instance.once (() => { return new Client (); });
     }
 }
+
+[Flags]
+private enum DriverFlags {
+    NVIDIA_LEGACY = 1 << 0,
+    NVIDIA_CURRENT = 1 << 1
+}
+
+// Gathers information about drivers that have been detected.
+//
+// We're using this to hide legacy NVIDIA drivers when a current driver is detected.
+private DriverFlags driver_flags (string[] drivers) {
+    DriverFlags flags = 0;
+
+    foreach (unowned string driver in drivers) {
+        if (driver.strip () == "") {
+            continue;
+        }
+        if (driver_is_nvidia_legacy(driver)) {
+            flags |= DriverFlags.NVIDIA_LEGACY;
+        } else if (driver.has_prefix("nvidia-driver")) {
+            flags |= DriverFlags.NVIDIA_CURRENT;
+        }
+    }
+
+    return flags;
+}
+
+private bool driver_is_nvidia_legacy (string driver) {
+    return driver == "nvidia-driver-390" || driver == "nvidia-driver-340";
```

Comment:
Do we want to hardcode these legacy versions in or just take the older of two listed versions as the legacy version? Would we ever see anything else listed from `ubuntu-drivers list`?

Reply:
We could simply ignore everything but the highest version available.

Reply:
Okay, the branch has been revised to only display the latest NVIDIA driver supported.